File: src/main/java/io/realworld/api/FavoritesApiImpl.java
package io.realworld.api;

import io.realworld.model.CreateArticle201Response;

import io.vertx.core.Future;
import io.vertx.ext.web.handler.HttpException;

// Implement this class

public class FavoritesApiImpl implements FavoritesApi {
    public Future<ApiResponse<CreateArticle201Response>> createArticleFavorite(String slug) {
        return Future.failedFuture(new HttpException(501));
    }

    public Future<ApiResponse<CreateArticle201Response>> deleteArticleFavorite(String slug) {
        return Future.failedFuture(new HttpException(501));
    }

}

File: src/main/kotlin/io/realworld/api/FavoritesApiImpl.kt
package io.realworld.api

import io.realworld.model.response.ApiResponse
import io.realworld.model.response.CreateArticle201Response
import io.vertx.core.Future
import io.vertx.ext.web.handler.HttpException

class FavoritesApiImpl:FavoritesApi {
  override fun createArticleFavorite(slug: String): Future<ApiResponse<CreateArticle201Response>> {
    return Future.failedFuture(HttpException(501))
  }

  override fun deleteArticleFavorite(slug: String): Future<ApiResponse<CreateArticle201Response>> {
    return Future.failedFuture(HttpException(501))
  }
}
